source/connection.c

```diff
@@ -49,9 +50,17 @@ struct aws_http_server {
     size_t initial_window_size;
     void *user_data;
     aws_http_server_on_incoming_connection_fn *on_incoming_connection;
+    aws_http_server_on_destroy_fn *on_destroy_complete;
 
     struct aws_hash_table channel_to_connection_map;
```

Comment:
we do not need to protect this hash table by lock, because if the sever is shutting down, the bool value will protect from more new connections to the hash table. If not, concurrent insertions and deletions will be fine.

Reply:
Because the original code do not protect the hash table, I though it was fine to do w/r concurrently. I was wrong